auth/src/test/java/com/firebase/ui/auth/viewmodel/SmartLockHandlerTest.java

```diff
@@ -86,20 +86,21 @@ public void testSaveCredentials_resolution() {
         mHandler.saveCredentials(TestHelper.getMockFirebaseUser(), TestConstants.PASSWORD, null);
 
         // Make sure we get a resolution
-        ArgumentCaptor<PendingResolution> resolveCaptor = ArgumentCaptor.forClass(PendingResolution.class);
-        verify(mResolutionObserver).onChanged(resolveCaptor.capture());
+        ArgumentCaptor<Resource<Void>> resolveCaptor = ArgumentCaptor.forClass(Resource.class);
+        verify(mResultObserver, times(2)).onChanged(resolveCaptor.capture());
```

Comment:
Should we change the test to actually verify the `onChanged` events separately?  Kind of confusing to verify that the second time is a `PendingIntentRequiredException` but we don't assert/care what the first event is.

Reply:
We do [below](https://github.com/firebase/FirebaseUI-Android/pull/1188/files/3acea2daf6aef96b14f3f99b22352b1433e5ee7b#diff-dc55ed1a9fc074c605f9a673fc79e118R98) but I got an exception without the `times(2)`. Is there a prettier way to do this with mockito?

Reply:
Just for learning purposes (I am fine with your code) you can tell Mockito you want to do events in order.

Here is some code from another project of mine which is basically the same (over there we call `Resource` a `Loadable`):

```java
    // Verify order of events
    InOrder inOrder = Mockito.inOrder(loadingObserver);

    // Start by loading
    liveData.postValue(Loadable.forLoading());
    inOrder.verify(loadingObserver).onChanged(true);

    // Move to success
    liveData.postValue(Loadable.forSuccess("Hello"));
    inOrder.verify(loadingObserver).onChanged(false);

    // Make sure that double-success does not make the observer get called again
    liveData.postValue(Loadable.forSuccess("This should have never been posted"));
    inOrder.verify(loadingObserver, never()).onChanged(anyBoolean());
```

Reply:
Whoa, that looks super cool!

> Just for learning purposes

Says you, I definitely using that! 😆 Thanks for tip 👍

Reply:
Want me to merge-as is or do you want to go and use `InOrder`?  Up to you.

Reply:
Sorry for the delay, my laptop died in the middle of running tests. 😭😆 Anyway, it's much prettier now!